test/extensions/clusters/aggregate/cluster_integration_test.cc

```diff
@@ -0,0 +1,285 @@
+#include "common/config/resources.h"
+#include "common/protobuf/protobuf.h"
+#include "common/protobuf/utility.h"
+
+#include "test/common/grpc/grpc_client_integration.h"
+#include "test/integration/http_integration.h"
+#include "test/integration/utility.h"
+#include "test/mocks/server/mocks.h"
+#include "test/test_common/network_utility.h"
+#include "test/test_common/simulated_time_system.h"
+#include "test/test_common/utility.h"
+
+#include "absl/synchronization/notification.h"
+#include "gtest/gtest.h"
+
+using testing::AssertionResult;
+
+namespace Envoy {
+namespace {
+
+const char ClusterName1[] = "cluster_1";
+const char ClusterName2[] = "cluster_2";
+const int UpstreamIndex1 = 2;
+const int UpstreamIndex2 = 3;
+
+const std::string config(const std::string& api_type) {
+  return fmt::format(
+      R"EOF(
+admin:
+  access_log_path: /dev/null
+  address:
+    socket_address:
+      address: 127.0.0.1
+      port_value: 0
+dynamic_resources:
+  cds_config:
+    api_config_source:
+      api_type: {}
+      grpc_services:
+        envoy_grpc:
+          cluster_name: my_cds_cluster
+      set_node_on_first_message_only: true
+static_resources:
+  clusters:
+  - name: my_cds_cluster
+    http2_protocol_options: {{}}
+    hosts:
+      socket_address:
+        address: 127.0.0.1
+        port_value: 0
+  - name: aggregate_cluster
+    connect_timeout: 0.25s
+    lb_policy: CLUSTER_PROVIDED
+    cluster_type:
+      name: envoy.clusters.aggregate
+      typed_config:
+        "@type": type.googleapis.com/envoy.config.cluster.aggregate.ClusterConfig
+        clusters:
+        - cluster_1
+        - cluster_2
+  listeners:
+    name: http
+    address:
+      socket_address:
+        address: 127.0.0.1
+        port_value: 0
+    filter_chains:
+      filters:
+        name: envoy.http_connection_manager
+        config:
+          stat_prefix: config_test
+          http_filters:
+            name: envoy.router
+          codec_type: HTTP2
+          route_config:
+            name: route_config_0
+            validate_clusters: false
+            virtual_hosts:
+              name: integration
+              routes:
+              - route:
+                  cluster: cluster_1
+                match:
+                  prefix: "/cluster1"
+              - route:
+                  cluster: cluster_2
+                match:
+                  prefix: "/cluster2"
+              - route:
+                  cluster: aggregate_cluster
+                match:
+                  prefix: "/aggregatecluster"
+              domains: "*"
+)EOF",
+      api_type);
+}
+
+class AggregateIntegrationTest : public Grpc::DeltaSotwIntegrationParamTest,
+                                 public HttpIntegrationTest {
+public:
+  AggregateIntegrationTest()
+      : HttpIntegrationTest(
+            Http::CodecClient::Type::HTTP2, ipVersion(),
+            config(sotwOrDelta() == Grpc::SotwOrDelta::Sotw ? "GRPC" : "DELTA_GRPC")) {
```

Comment:
we can probably skip this, i dont think the xds update mechanism is relevant to how this cluster works